write_asv_machine.py

```diff
@@ -0,0 +1,32 @@
+import copy
+import hashlib
+import json
+import os
+
+from asv.machine import Machine, MachineCollection
+
+
+def write_machine_info():
+    # Hack making sure that similar machines will have same name, so that
+    # we can reuse results from previous runs.
+    if os.path.exists(MachineCollection.get_machine_file_path()):
+        os.remove(MachineCollection.get_machine_file_path())
+
+    machine_info = Machine.get_defaults()
+    # approx ram size to MB
+    machine_info["ram"] = machine_info["ram"] // int(1e3)
+
+    machine_marker = copy.copy(machine_info)
+    del machine_marker["machine"]
+
+    marker_str = json.dumps(machine_marker, sort_keys=True)
+    suffix = hashlib.md5(marker_str.encode("utf-8")).hexdigest()[:5]
+
+    machine_name = "machine_{}".format(suffix)
+    machine_info["machine"] = machine_name
+
+    MachineCollection.save(machine_name, machine_info)
+
+
+if __name__ == "__main__":
+    write_machine_info()
```

Comment:
So what runs this? Pytest test?

Reply:
I did not include execution script in github actions yaml, thanks for pointing out!